SingularityUI/app/models/Request.coffee

```diff
@@ -427,69 +434,120 @@ class Request extends Model
                 return unless confirmed
                 @unpause(confirmed).done callback
 
-    promptRun: (callback) =>
-        try
-            lastCommands = JSON.parse(localStorage.getItem(@localStorageCommandLineInputKeyPrefix + @id))
-        catch e
-            console.error('Could not parse previous commands JSON')
-            lastCommands = []
-        vex.dialog.prompt
-            message: "<h3>Run Task</h3>"
-            input: runTemplate
-                id: @get "id"
-                prefix: @localStorageCommandLineInputKeyPrefix
-                commands: if lastCommands? then lastCommands[lastCommands.length - 1] else []
+    getMostRecentlyRunTask: (callback) =>
+        @commands = []
+        historicalTasks = new HistoricalTasks [],
+            params:
+                requestId: @get "id"
+        historicalTasksFetch = historicalTasks.fetch()
+        historicalTasksFetch.success () =>
+            if historicalTasks.models.length
+                taskId = historicalTasks.models[0].attributes.taskId.id
+                task = new TaskHistory {taskId}
+                taskFetch = task.fetch()
+                taskFetch.success () =>
+                    @commands = task.attributes.task.taskRequest.pendingTask.cmdLineArgsList
+                    callback()
+                taskFetch.error () =>
+                    app.caughtError() # Don't actually error if we can't find previous args, just don't populate
+                    callback()
+            else
+                callback() # If this is the first task being run for this reqeust
+        historicalTasksFetch.error () =>
+            app.caughtError()
+            callback() # Don't actually error if we can't find previous args, just don't populate
+
+    promptRun: (callback, task) => #task is an optional parameter - if it's provided this will rerun it, else this will run a new task
+        showDialog = () =>
+            if task
+                commands = task.attributes.task.taskRequest.pendingTask.cmdLineArgsList
+            else
+                commands = @commands
+            vex.dialog.prompt
+                message: "<h3>#{if task then 'Rerun' else 'Run'} Task</h3>"
+                input: runTemplate
+                    id: @get "id"
+                    prefix: @localStorageCommandLineInputKeyPrefix
```

Comment:
do we still need this prefix if we aren't using localStorage for populating args anymore?